packages/cinterop/src/jvmMain/CMakeLists.txt

```diff
@@ -4,18 +4,40 @@ find_package(JNI)
 set(SWIG_JNI_GENERATED "${CMAKE_SOURCE_DIR}/../../../jni-swig-stub/build/generated/sources/jni")
 set(SWIG_JNI_HELPERS "${CMAKE_SOURCE_DIR}/../../../jni-swig-stub/src/main/jni")
 set(CINTEROP_JNI "${CMAKE_SOURCE_DIR}/../jvm/jni")
+set(REALM_TARGET_LINK_LIBS RealmFFIStatic Realm::ObjectStore)
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17")
+if (CMAKE_SYSTEM_NAME MATCHES "^Windows")
+    MESSAGE("Building JNI for Windows")
+    set(CMAKE_CXX_STANDARD 17)
```

Comment:
Is this the only way to set `-std=c++17"`on Windows? It would have been nice if we could have move `set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17")` outside all the targets